### PR TITLE
LPS-197807 Return false instead of null when the user does not have permission to view the content

### PR DESCRIPTION
**Pull request by vngoliveira**
LPP ticket: [LPP-50732](https://liferay.atlassian.net/browse/LPP-50732)
Related issue: [LPS-197807](https://liferay.atlassian.net/browse/LPS-197807)

### CODE REVIEW

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:workflow

**Comment by mateussgn** [comment]
ci:test:relevant

File: portal-kernel/src/com/liferay/portal/kernel/workflow/permission/WorkflowPermissionUtil.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ private static Boolean _hasPermission(
 				permissionChecker, workflowInstance);
 
 			if (!hasPermission && actionId.equals(ActionKeys.VIEW)) {
-				return null;
+				return false;
```
Comment (mateussgn) [inline portal-kernel/src/com/liferay/portal/kernel/workflow/permission/WorkflowPermissionUtil.java:104]: @vngoliveira The method signature requires a Boolean Java Object,
Try to use like in line [89](https://github.com/liferay/liferay-portal/blob/e1c1485ab3210fdcd5abab6226c295bf3617a14a/portal-kernel/src/com/liferay/portal/kernel/workflow/permission/WorkflowPermissionUtil.java#L89) or line [75](https://github.com/liferay/liferay-portal/blob/e1c1485ab3210fdcd5abab6226c295bf3617a14a/portal-kernel/src/com/liferay/portal/kernel/workflow/permission/WorkflowPermissionUtil.java#L75).
Would look like this:
```suggestion
				return Boolean.FALSE;
```

**Review by mateussgn — CHANGES_REQUESTED** [review]
Just a slightly change.

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:workflow

**Review by mateussgn — APPROVED** [review]
Good job!

**Review by guilhermedcamacho — APPROVED** [review]
LGTM :tada:

**Comment by guilhermedcamacho** [comment]
ci:forward

**Comment by guilhermedcamacho** [comment]
Hey @vngoliveira, have you tested if this change actually fixes the bug?
According to Lourde's [comment](https://liferay.atlassian.net/browse/LPS-197807?focusedCommentId=2486141) on the ticket, the old code was introduced by this [bug](https://liferay.atlassian.net/browse/LPSA-57484), so it would also be good to test whether this scenario is still covered.

**Comment by vngoliveira** [comment]
Hey @guilhermedcamacho , I tested the mentioned ticket and both issues were fixed by this alteration. Can we go ahead with this solution?

**Comment by guilhermedcamacho** [comment]
> Hey @guilhermedcamacho , I tested the mentioned ticket and both issues were fixed by this alteration. Can we go ahead with this solution?

Sure, thanks!

**Comment by guilhermedcamacho** [comment]
Manually forwarded to [brianchandotcom#142852](https://github.com/brianchandotcom/liferay-portal/pull/142852)

**Pull request closed without merging**